### PR TITLE
Refactor of components strucutre

### PR DESCRIPTION
**Pull request by ironsidevsquincy**
Yikes! Sorry. Top-level changes are

 * Make the card components more content type agnostic/remove logic from it, i.e. just supplied the data and let it do it's thing
 * Now, the `Content` component handles getting the data in a workable format and passing it to the cards
 * Three card types now, `Article`, `Concept` and `Video` (might be able to consolidate these down in the future
 * shared components/libs now live in `/shared/`
 * renamed css/js files to match their container directory

### CODE REVIEW

File: config/layouts/myft.js [hunk 1]
```diff
@@ -26,8 +24,7 @@ export default [
                 colspan: { default: 12, M: 4, L: 3 },
                 components: [
                     {
-                        type: Card,
-                        cardType: Concept
+                        type: Content
```
Comment (adgad) [inline config/layouts/myft.js]: Should these be Concept?

Comment (adgad) [inline config/layouts/myft.js]: Oh never mind, think I misread your description

File: shared/components/section/section-content/section-content.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+
+import Content from '../../content/content';
+
+// assign an incremental id to the Content components
+const assignContentId = (contentIndex, component) => {
+	if (component.type === Content) {
+		component.id = contentIndex++;
+	} else if (component.components && component.components.length) {
+		contentIndex = component.components.reduce(assignContentId, contentIndex);
```
Comment (adgad) [inline shared/components/section/section-content/section-content.js]: Does this handle the mostPopular with ad section (where the index had to be specified in the layout?)

Comment (adgad) [inline shared/components/section/section-content/section-content.js]: Also maybe worth checking/testing for missing content (i.e. @matthew-andrews fix from the weekend)

Comment (ironsidevsquincy) [inline shared/components/section/section-content/section-content.js]: yes: https://github.com/Financial-Times/next-front-page/pull/518/files#diff-b3564207f65a74810d4c8fcd13ae03a9R72
and yes: https://github.com/Financial-Times/next-front-page/pull/518/files#diff-b3564207f65a74810d4c8fcd13ae03a9R73

**Comment by adgad** [comment]
:+1:

**Pull request merged**